### PR TITLE
dev: Add tracing

### PR DESCRIPTION
**Pull request by abbyoung**
# SC-64

## Proposed changes

* Adds `tracing.js` which initializes open telemetry/xray tracing.
* Implements auto-instrumentation using the node and web metapackages (ALL THE THINGS)

---

## Setup

* When branch is deployed to dev, you should be able to see _lots_ of logs in X-Ray tracing.

---

### As the original developer, I have

- [x] Met the acceptance criteria, or will meet them in subsequent PRs or stories

### As code reviewer(s), I have

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] Considered marking this as accepted even if there are small changes needed

---

### CODE REVIEW

File: startup/tracing.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+// OTEL tracing
+const { BatchSpanProcessor } = require('@opentelemetry/tracing')
+const { NodeTracerProvider } = require('@opentelemetry/node')
+const { OTLPTraceExporter } = require('@opentelemetry/exporter-trace-otlp-grpc')
+const { AWSXRayPropagator } = require('@opentelemetry/propagator-aws-xray')
+const { AWSXRayIdGenerator } = require('@opentelemetry/id-generator-aws-xray')
+const { trace } = require('@opentelemetry/api')
+
+const { detectResources } = require('@opentelemetry/resources')
+const { awsEc2Detector } = require('@opentelemetry/resource-detector-aws')
```
Comment (esacteksab) [inline startup/tracing.js:11]: Should this be `awsECSDetector` instead? Also, line 16 below?

File: package.json [hunk 2]
```diff
@@ -43,6 +43,17 @@
...
+    "@opentelemetry/auto-instrumentations-node": "^0.27.4",
+    "@opentelemetry/exporter-trace-otlp-grpc": "^0.27.0",
+    "@opentelemetry/id-generator-aws-xray": "^1.0.1",
+    "@opentelemetry/instrumentation-aws-sdk": "^0.5.1",
+    "@opentelemetry/node": "^0.24.0",
+    "@opentelemetry/propagator-aws-xray": "^1.0.1",
+    "@opentelemetry/resource-detector-aws": "^1.0.3",
+    "@opentelemetry/resources": "^1.0.1",
+    "@opentelemetry/sdk-trace-node": "^1.0.1",
+    "@opentelemetry/tracing": "^0.24.0",
```
Comment (suzubara) [inline package.json:56]: can we pin all of the above pls?

File: src/pages/api/sysinfo.ts [hunk 3]
```diff
@@ -1,16 +1,55 @@
...
+    return JSON.stringify({ traceId: xrayTraceId })
+  }
+  return
+}
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  console.log('Responding to /sysinfo')
+  const tracer = await tracing.initTracing()
```
Comment (suzubara) [inline src/pages/api/sysinfo.ts:32]: does this need to be init again if it happened at startup?

File: package.json (hunk 2 above)
Comment (abbyoung) [reply package.json:57]: YES this was just from testing -- thank you for the reminder!

File: src/pages/api/sysinfo.ts (hunk 3 above)
Comment (abbyoung) [reply src/pages/api/sysinfo.ts]: all this code was deleted since it was just for testing the hookup cause no, it doesn't

**Review by suzubara — APPROVED** [review]

**Pull request merged**